presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoRpmControlUtils.java

```diff
@@ -0,0 +1,249 @@
+                        LOGGER.error("Coordinator can't be gracefully stopped.");
+                        return Response.status(CONFLICT).build();
+                    }
+                    gracefulStop();
+                    break;
+                default:
+                    LOGGER.error("Invalid stop type: {}", stopType);
+                    return Response.status(INTERNAL_SERVER_ERROR).build();
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    public static Response restartUsingRpm()
+    {
+        try {
+            if (!isInstalled()) {
+                LOGGER.error("Presto is not installed");
+                return Response.status(NOT_FOUND).build();
+            }
+            int prestoRestart = executeCommand(SUBPROCESS_TIMEOUT, "service", "presto", "restart");
+            if (prestoRestart != 0) {
+                throw new PrestoManagerException("Failed to restart presto", prestoRestart);
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    public static Response statusUsingRpm()
+    {
+        try {
+            if (!isInstalled()) {
+                LOGGER.info("Presto is not installed");
+                return Response.status(OK).entity(GSON.toJson("Presto is not installed"))
+                        .type(APPLICATION_JSON).build();
+            }
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        try {
+            String prestoPort = getPrestoPort();
+            UriBuilder uriBuilder = fromUri(format("http://localhost:%s", prestoPort)).path("/v1/info");
+            Response prestoInfo = JERSEY_CLIENT.target(uriBuilder.build()).request(APPLICATION_JSON).buildGet().invoke();
+            Response prestoState = JERSEY_CLIENT.target(uriBuilder.path("/state").build())
+                    .request(APPLICATION_JSON).buildGet().invoke();
+            JsonParser jsonParser = new JsonParser();
+            JsonObject prestoStatus = jsonParser.parse(prestoInfo.readEntity(String.class)).getAsJsonObject();
+            prestoStatus.addProperty("state", jsonParser.parse(prestoState.readEntity(String.class)).getAsString());
+            return Response.status(OK).entity(prestoStatus.toString()).type(APPLICATION_JSON).build();
+        }
+        catch (ProcessingException e) {
+            LOGGER.info("Presto is not running", e);
+            return Response.status(OK).entity(GSON.toJson("Presto is installed but not running"))
```

Comment:
Java doesn't do well with pipes. 
So querying rpm is not an option. 
There's a hack to this:
We can query the presto version `rpm -qi presto-server-rpm` ,  then save the output. Parse the output and get the presto version.